build/media_source/com_templates/vue/components/settings/editelement.vue

```diff
@@ -0,0 +1,203 @@
+                                <td class="control-group">
+                                    <select name="element_offset" id="element_offset" class="custom-select custom-select-sm" v-model="element_offset.lg">
+                                        <option v-for="opt in offset" v-if="opt.value <= threshold" :value="opt.value">
+                                            {{opt.label}}
+                                        </option>
+                                    </select>
+                                </td>
+                            </tr>
+                        </tbody>
+                    </table>
+                </div>
+                <div v-if="elementSelected.type === 'moduleposition'">
+                    <label for="moduleposition_name">{{ translate('COM_TEMPLATES_POSITION_NAME') }}</label>
+                    <input type="text" name="moduleposition_name" id="moduleposition_name" class="form-control" v-model="element_moduleposition_name">
+                    <br>
+                    <label for="module_chrome">{{ translate('COM_TEMPLATES_SELECT_MODULE_CHROME') }}</label>
+                    <select class="custom-select" id="module_chrome" name="module_chrome" v-model="element_module_chrome">
+                        <option value="none">{{ translate('COM_TEMPLATES_NONE') }}</option>
+                        <option value="rounded">{{ translate('COM_TEMPLATES_ROUNDED') }}</option>
+                        <option value="table">{{ translate('COM_TEMPLATES_TABLE') }}</option>
+                        <option value="horz">{{ translate('COM_TEMPLATES_HORZ') }}</option>
+                        <option value="xhtml">{{ translate('COM_TEMPLATES_XHTML') }}</option>
+                        <option value="html5">{{ translate('COM_TEMPLATES_HTML5') }}</option>
+                        <option value="outline">{{ translate('COM_TEMPLATES_OUTLINE') }}</option>
+                    </select>
+                </div>
+            </div>
+            <div>
+                <button type="button" class="btn btn-success" @click="modifyElement">{{ translate('COM_TEMPLATES_ADD') }}</button>
+                <button type="button" class="btn btn-secondary" @click="closeNav">{{ translate('JTOOLBAR_CLOSE') }}</button>
+            </div>
+        </fieldset>
+    </div>
+</template>
+
+<script>
+import { mapState, mapMutations } from 'vuex';
+
+export default {
+    name: 'edit-element',
+    computed: {
+        ...mapState([
+            'elementSelected',
+            'parent'
+        ]),
+        threshold() {
+            return this.gridSize - this.elementSelected.options.size;
+        }
+    },
+    watch: {
+        elementSelected: {
+            deep: true,
+            handler() {
+                this.element_class = this.elementSelected.options.class;
+                this.element_module_chrome = this.elementSelected.options.module_chrome ? this.elementSelected.options.module_chrome : 'none';
```

Comment:
Are you sure that `this.elementSelected.options.class` value always be present? If options are not present then it will throw error.

Reply:
We have added "class" in JSON corresponding to every element